be/src/runtime/snapshot_loader.cpp

```diff
@@ -673,6 +679,44 @@ Status SnapshotLoader::move(const std::string& snapshot_path, const TabletShared
             VLOG(2) << "link file from " << full_src_path << " to " << full_dest_path;
         }
 
+        if (dcg_file.size() != 0) {
+            DeltaColumnGroupSnapshotPB dcg_snapshot_pb;
+            std::string full_path = snapshot_path + "/" + dcg_file;
+            auto st = DeltaColumnGroupListHelper::parse_snapshot(full_path, dcg_snapshot_pb);
+            if (!st.ok()) {
+                return Status::InternalError("failed to parse dcgs meta");
+            }
+
+            if (dcg_snapshot_pb.dcg_lists().size() != 0) {
+                int idx = 0;
+                auto data_dir = tablet->data_dir();
+                rocksdb::WriteBatch wb;
+
+                RETURN_IF_ERROR(TabletMetaManager::clear_delta_column_group(data_dir, &wb, tablet_id));
+
+                for (auto& dcg_list_pb : dcg_snapshot_pb.dcg_lists()) {
+                    // dcgs for each segment
+                    DeltaColumnGroupList dcgs;
+                    RETURN_IF_ERROR(
+                            DeltaColumnGroupListSerializer::deserialize_delta_column_group_list(dcg_list_pb, &dcgs));
+
+                    if (dcgs.size() == 0) {
+                        continue;
+                    }
+
+                    // tablet id and rowsetid in PB have been replace by the new one here.
+                    RETURN_IF_ERROR(TabletMetaManager::put_delta_column_group(
+                            data_dir, &wb, dcg_snapshot_pb.tablet_id(idx), dcg_snapshot_pb.rowset_id(idx),
+                            dcg_snapshot_pb.segment_id(idx), dcgs));
+                    ++idx;
+                }
+                st = data_dir->get_meta()->write_batch(&wb);
+                if (!st.ok()) {
+                    return Status::InternalError("save dcgs meta failed");
```

Comment:
Print the tablet_id in the error log to better trace.

Reply:
done